name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/MerkurPrivatBankPDFExtractor.java

```diff
@@ -0,0 +1,115 @@
+
+@SuppressWarnings("nls")
+public class MerkurPrivatBankPDFExtractor extends AbstractPDFExtractor
+{
+
+    public MerkurPrivatBankPDFExtractor(Client client)
+    {
+        super(client);
+        
+        // @formatter:off
+        // Am Marktplatz 10 · 97762 Hammelburg
+        // @formatter:on
+        addBankIdentifier("Am Marktplatz 10 · 97762 Hammelburg"); //$NON-NLS-1$
+
+        addBuySellTransaction();
+    }
+
+    @Override
+    public String getLabel()
+    {
+        return "MERKUR PRIVATBANK KGaA"; //$NON-NLS-1$
+    }
+
+    private void addBuySellTransaction()
+    {
+        DocumentType type = new DocumentType("Kauf");
+        this.addDocumentTyp(type);
+
+        Transaction<BuySellEntry> pdfTransaction = new Transaction<>();
+        pdfTransaction.subject(() -> {
+            BuySellEntry entry = new BuySellEntry();
+            entry.setType(PortfolioTransaction.Type.BUY);
+            return entry;
+        });
+        
+        // Start from: Wertpapier Abrechnung Kauf
+        Block firstRelevantLine = new Block("^Wertpapier Abrechnung Kauf.*$");
+        type.addBlock(firstRelevantLine);
+        firstRelevantLine.set(pdfTransaction);
+        
+        pdfTransaction
+                        // @formatter:off
+                        // Type (Optional)
+                        // Wertpapier Abrechnung Kauf
+                        // @formatter:on
+                        .section("type").optional().match("^Wertpapier Abrechnung (?<type>Kauf).*$").assign((t, v) -> {
+                            if (v.get("type").equals("Kauf"))
+                                t.setType(PortfolioTransaction.Type.BUY);
+                        })
+                        // @formatter:off
+                        // Security identification:
+                        // Stück 125,3258 XTR.(IE) - MSCI WORLD              IE00BJ0KDQ92 (A1XB5U)
+                        // REGISTERED SHARES 1C O.N.
+                        // @formatter:on
+                        .section("name", "ln2", "isin", "wkn")
+                        .match("^St.ck [\\,\\d]+ (?<name>.*) (?<isin>[A-Z]{2}[A-Z0-9]{9}[0-9]) \\((?<wkn>[A-Z0-9]{6})\\)$")
+                        .match("^(?<ln2>.*)$").assign((t, v) -> {
+                            // Assemble name from both lines
+                            if (!v.get("ln2").startsWith("Handels-/Ausführungsplatz"))
+                                v.put("name", trim(v.get("name")) + " " + trim(v.get("ln2")));
+                            t.setSecurity(getOrCreateSecurity(v));
+                        })
+                        // @formatter:off
+                        // Shares of the transaction
+                        // Stück 125,3258
+                        // @formatter:on
+                        .section("shares").match("^St.ck (?<shares>[\\,\\d]+) .*$").assign((t, v) -> 
+                            t.setShares(asShares(v.get("shares")))
+                        )
```

Comment:
Formatting Source.
```
.section("shares").match("^St.ck (?<shares>[\\,\\d]+) .*$").assign((t, v) -> 
                            t.setShares(asShares(v.get("shares")))
                        )
```
to this.
```
.section("shares")
.match("^St.ck (?<shares>[\\,\\d]+) .*$")
.assign((t, v) -> t.setShares(asShares(v.get("shares"))))
```
Cleanly written code simplifies later maintenance.